### ISSUE TITLE
E1011 on FindInMap with DefaultValue, when without Transform

### ISSUE THREAD

**Issue by EvanK** [aws-cloudformation/cfn-lint#4627]
### CloudFormation Lint Version

1.55.0

### What operating system are you using?

Windows (Git-bash)

### Describe the bug

When linting a template that uses the `FindInMap` intrinsic function with a `DefaultValue` fourth element, it reports the following error by default:

> `[cfn-lint] E1011: expected maximum item count: 3, found: 4`

The error goes away when the template uses `Transform: AWS::LanguageExtensions`, but according to [AWS "Fn::FindInMap enhancements" docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/intrinsic-function-reference-findinmap-enhancements.html), `AWS::LanguageExtensions` is no longer required.

### Expected behavior

Expected the template to validate a proper use of `FindInMap`, without the need for `Transform: AWS::LanguageExtensions`

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
# Transform: AWS::LanguageExtensions
Parameters:
  EnvironmentType:
    Description: The environment type (Dev or Prod)
    Type: String
    Default: Dev
    AllowedValues:
      - Dev
      - Prod

Mappings:
  SecurityGroups:
    Dev:
      SecurityGroupIds: sg-12345678
    Prod:
      SecurityGroupIds: sg-abcdef01,sg-ghijkl23

Resources:
  SomeFunction:
    Type: AWS::Lambda::Function
    Properties:
      Environment:
        Variables:
          MAPPED_VALUE:
            Fn::FindInMap:
            - SecurityGroups
            - !Ref EnvironmentType
            - SecurityGroupIds
            - DefaultValue: 'nevermind'
      Code:
        ZipFile: |
          export function handler(event,context) {
            console.dir({ event, context });
          }
      Handler: index.handler
      Runtime: nodejs24.x
      Role: arn:aws:iam::123456789012:role/someRoleName
```
